src/core/bpf-firewall.c

```diff
@@ -392,7 +429,16 @@ static int bpf_firewall_prepare_access_maps(
                 if (!cc)
                         continue;
 
-                bpf_firewall_count_access_items(verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny, &n_ipv4, &n_ipv6);
+                list = verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny;
+
+                bpf_firewall_count_access_items(list, &n_ipv4, &n_ipv6);
+
+                /* Skip making the LPM trie map in cases where we are using "any" in order to hack around
+                 * needing CAP_SYS_ADMIN for allocating LPM trie map. */
+                if (ip_address_access_item_is_any(list)) {
+                        *has_any = true;
+                        return 0;
+                }
```

Comment:
we usually follow the rule that return parameters should be strictly initialized on success of a function and left untouched in case of failure. However, in this function *has_any is left untouched also in some cases of success. Hence, please, further down, at the end of this function, could you explicitly set `*has_any = false;`, so that our rule holds again? I think it's highly valuable if we follow common semantics on return parameters here.